ui/src/containers/CreateVolume.js

```diff
@@ -421,6 +421,33 @@ const CreateVolume = (props) => {
                 return items.find((item) => item.value === selectedValue);
               };
 
+              //if re-check the box again, we should only update/pre-fill the defaults for unchanged field.
+              // to make sure to keep the user customization.
+              const handleCheckboxChange = (field) => (selectedObj) => {
+                setFieldValue(field, selectedObj.target.checked);
+                // Clear the untouched field to get the update from the global value
+                for (let i = 0; i < values.numberOfVolumes; i++) {
+                  if (
+                    touched &&
+                    touched.volumes &&
+                    touched?.volumes[i]?.name === true
+                  ) {
+                    continue;
+                  } else {
+                    setFieldValue(`volumes[${i}]name`, '');
+                  }
+                  if (
+                    touched &&
+                    touched.volumes &&
+                    touched?.volumes[i]?.path === true
+                  ) {
+                    continue;
+                  } else {
+                    setFieldValue(`volumes[${i}]path`, '');
+                  }
```

Comment:
I think this can be simplified to : 
```js
if (!touched?.volumes[i]?.name) {
    setFieldValue(`volumes[${i}]name`, '');
}
if (!touched?.volumes[i]?.path) {
    setFieldValue(`volumes[${i}]path`, '');
}
```

Reply:
Hi @JBWatenbergScality ,

Yes. Good point. But I guess we need to use optional chaining to make sure the Volumes array exists.
Please see:

```js
                  if (!touched?.volumes?.[i]?.name) {
                    setFieldValue(`volumes[${i}]name`, '');
                  }
                  if (!touched?.volumes?.[i]?.path) {
                    setFieldValue(`volumes[${i}]path`, '');
                  }
```